File: cpp/41_xenia_and_ringroad.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    long long n, m;
    if (!(cin >> n >> m))
        return 0;
    long long cur = 1, ans = 0;
    for (int i = 0; i < m; i++)
    {
        long long t;
        cin >> t;
        if (t >= cur)
            ans += t - cur;
        else
            ans += n - (cur - t);
        cur = t;
    }
    cout << ans << "\n";
}

File: python/41_xenia_and_ringroad.py
n, m = map(int, input().split())
cur = 1
ans = 0
for _ in range(m):
    t = int(input().strip())
    if t >= cur:
        ans += t - cur
    else:
        ans += n - (cur - t)
    cur = t
print(ans)
